sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosTable.scala

```diff
@@ -97,4 +105,21 @@ class CosmosTable(val transforms: Array[Transform],
       containerStateHandle
     )
   }
+
+  // This can be used only when databaseName and ContainerName are specified.
+  private[spark] def initializeAndBroadcastCosmosClientStateForContainer(): Broadcast[CosmosClientMetadataCachesSnapshot] = {
+    try {
+      client.getDatabase(cosmosContainerConfig.database).getContainer(cosmosContainerConfig.container).readItem(
+        UUID.randomUUID().toString, new PartitionKey(UUID.randomUUID().toString), classOf[ObjectNode])
+        .block()
+    } catch {
+      case _: CosmosException => None
+    }
+
+    val state = new CosmosClientMetadataCachesSnapshot()
+    state.serialize(client)
+
+    val sparkSession = SparkSession.active
+    sparkSession.sparkContext.broadcast(state)
```

Comment:
When should we do this broadcast of the state? Asking to understand what's the pattern. I've seen in other places the opposite, basically wire the state into the builder.

Reply:
This is the original place where the broadcast happens - just got re-arranged